src/include/sof/coherent.h

```diff
@@ -78,9 +96,6 @@ __must_check static inline struct coherent *coherent_acquire(struct coherent *c,
 		CHECK_COHERENT_CORE(c);
 
 		spin_lock(&c->lock);
-
-		/* invalidate local copy */
-		dcache_invalidate_region(uncache_to_cache(c), size);
```

Comment:
Why are you removing this ? Our local cache may not be up to date with any changes other cores have written back.

Reply:
We still need to invalidate local L1 as it has know way of knowing uncache has changed since our last usage. e.g.
```core 0 -> core 1 -> core 0.```
In the above core 0 local L1 needs to be invalidated so it can be coherent with whatever core1 wrote back to uncache.

Reply:
@lgirdwood don't think so. We invalidate on release. When core 0 in your example last time released the memory, it invalidated its cache. It has no cached data now for this memory.

Reply:
ok, your right - but I'm not going to merge this as it requires a ton of validation. I'm happy to accept a comment here though with a TODO.